### PR TITLE
feat(services): migrate storage setting react query

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-920

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Write comments that add new information that doesn’t exist in the code

File: libs/shared/util-services/src/lib/build-edit-service-payload.ts [hunk 1]
```diff
@@ -133,7 +133,7 @@ function refactoContainer({ service: container, request = {} }: containerProps):
     memory: container.memory,
     max_running_instances: container.max_running_instances,
     min_running_instances: container.min_running_instances,
-    registry_id: container.registry_id || '',
+    registry_id: container.registry_id || container.registry?.id || '',
```
Comment (RemiBonnet) [inline libs/shared/util-services/src/lib/build-edit-service-payload.ts:136]: I have a special case here, that's why I added `container.registry?.id`, I think Container detail return an `registry` object

File: libs/pages/application/src/lib/feature/page-settings-storage-feature/storage-modal-feature/storage-modal-feature.tsx [hunk 2]
```diff
@@ -1,104 +1,58 @@
...
+import { buildEditServicePayload } from '@qovery/shared/util-services'
 import StorageModal from '../../../ui/page-settings-storage/storage-modal/storage-modal'
 
 export interface StorageModalFeatureProps {
-  storage?: ServiceStorageStorageInner
-  application?: ApplicationEntity
-  organizationId: string
-  projectId: string
-  applicationId: string
+  service: Extract<AnyService, Application | Container>
```
Comment (ctjhoa) [inline libs/pages/application/src/lib/feature/page-settings-storage-feature/storage-modal-feature/storage-modal-feature.tsx:10]:
```suggestion
  service: Application | Container
```

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/pages/application/src/lib/feature/page-settings-storage-feature/storage-modal-feature/storage-modal-feature.tsx (hunk 2 above)
Comment (RemiBonnet) [reply libs/pages/application/src/lib/feature/page-settings-storage-feature/storage-modal-feature/storage-modal-feature.tsx]: https://github.com/Qovery/console/pull/1097/commits/4a9b27bdd92080b60e252a2402f58e2946751eee

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**